backend/core/views.py

```
@@ -268,8 +269,89 @@ def update(self, request: Request, *args, **kwargs) -> Response:
         return super().update(request, *args, **kwargs)
 
+    def _validate_related_objects_access(self, request: Request, instance=None):
+        folder_id = request.data.get("folder")
+        if (
+            not folder_id
+            and "perimeter" in request.data
+            and request.data["perimeter"] not in (-1, None)
+        ):
+            try:
+                perimeter_id = uuid.UUID(str(request.data["perimeter"]))
+                perimeter = get_object_or_404(Perimeter, id=perimeter_id)
```

Ruff (0.8.2)

`Perimeter` may be undefined, or defined from star imports

(F405)